### DISCUSSION TITLE
Get all branch protection rules for repo

### DISCUSSION THREAD

**Discussion by 91dave** [community/community#24596] [Apps, API and Webhooks]
<p>Our org has ~100 repos, and we use branch protection rules, both for our <code>main</code> branch, and for feature branches (naming scheme: <code>feature/*</code>).</p>
<p>I’d like to write a script using <code>gh api</code> that iterates through all BP rules accross all repos, and prints out the ones that don’t match certain rules (specifically I want to list out any where rules are not enforced for Admins!)</p>
<p>I can easily enough do this for the main branch which always exists, using:</p>
<pre><code class="lang-auto">gh api repos/my-org/my-repo/branches/main/protection
</code></pre>
<p>The problem is, you can only GET a BP rule if there is currently a matching branch. Since we’re only ever working on a handful or repos at once, it means that ~90% of our repos (i.e. which do NOT have a branch matching <code>feature/*</code>) can’t be scanned for broken BP rules.</p>
<p>Is there any way around this with the GitHub CLI? Or possibly any language SDKs that would allow me to do this?</p>

**Comment by jsoref** [comment]
<p>You could cheat and have it create such a branch :wink: (<code>feature/verify-branch-protection</code>).</p>
<p>It could even just be equivalent to <code>main</code>.</p>

**Comment by robert-spring** [comment]
<p>I know this is old, but did you ever figure out a way to do this that doesn’t require creating a dummy branch?</p>

**Comment by 91dave** [comment, marked as answer]
<p>Yes! I ended up using the graphQL functionality. I am very much NOT a graphQL expert, however the following worked for me.</p>
<blockquote>
<p>Note that this is a bash script using the <code>gh</code> CLI tool to make API calls (which may not be everyone’s first choice of approach…)</p>
</blockquote>
<pre data-code-wrap="sh"><code class="lang-nohighlight">#!/bin/bash

owner=user_name
repo=my_repo

gh api graphql -f query="{
			repository(owner:\"$owner\", name:\"$repo\") { 
				branchProtectionRules(first:100){
					nodes{pattern,dismissesStaleReviews,isAdminEnforced,allowsDeletions,databaseId}
		}}}
</code></pre>
<p>The <code>nodes</code> bit is basically just selecting the fields you want. I suspect omitting this will return everything (but see above - not a Graph QL expert!).</p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="91dave" data-post="4" data-topic="223786">
<div class="title">
<div class="quote-controls"></div>
 91dave:</div>
<blockquote>
<p>The <code>nodes</code> bit is basically just selecting the fields you want. I suspect omitting this will return everything</p>
</blockquote>
</aside>
<p>With GraphQL you have to specify the data (fields) you want returned so omitting will not return all fields, glad you resolved your problem though</p>
